python/cuml/solvers/cd_mg.pyx

```diff
@@ -0,0 +1,536 @@
+
+        return arr_interfaces_trans
+
+    def fit_colocated(self, input_data, M, N, partsToRanks, rnk):
+        """
+        Fit function for MNMG Linear Regression. 
+        This not meant to be used as
+        part of the public API.
+        :param X: array of local dataframes / array partitions
+        :param M: total number of rows
+        :param N: total number of cols
+        :param partsToRanks: array of tuples in the format: [(rank,size)]
+        :return: self
+        """
+
+        arr_interfaces = []
+        arr_interfaces_y = []
+
+        for i in range(len(input_data)):
+            X_m, input_ptr, n_rows, self.n_cols, self.dtype = \
+                input_to_dev_array(input_data[i][0], check_dtype=[np.float32, np.float64])
+            arr_interfaces.append({"obj": X_m,
+                                   "data": input_ptr,
+                                   "shape": (n_rows, self.n_cols)})
+
+            y_m, input_ptr, n_rows, n_cols, self.dtype = \
+                input_to_dev_array(input_data[i][1], check_dtype=[np.float32, np.float64])
+            arr_interfaces_y.append({"obj": y_m,
+                                   "data": input_ptr,
+                                   "shape": (n_rows, n_cols)})
+            
+        
+        n_total_parts = len(input_data)
+        
+        cdef RankSizePair **rankSizePair = <RankSizePair**> \
+            malloc(sizeof(RankSizePair**)
+                   * n_total_parts)
+
+        for i in range(len(input_data)):
+            rankSizePair[i] = <RankSizePair*> \
+                    malloc(sizeof(RankSizePair))
+            rankSizePair[i].rank = <int>rnk
+            rankSizePair[i].size = <size_t>len(input_data[i][0])
+                        
+        self.coef_ = cudf.Series(zeros(self.n_cols,
+                                       dtype=self.dtype))
+        cdef uintptr_t coef_ptr = get_cudf_column_ptr(self.coef_)
+
+        cdef float c_intercept1
+        cdef double c_intercept2
+        cdef cumlHandle* handle_ = <cumlHandle*><size_t>self.handle.getHandle()
+        cdef uintptr_t data
+        cdef uintptr_t labels
+        
+        if self.dtype == np.float32:
+            data = self._build_dataFloat(arr_interfaces)
+            labels = self._build_dataFloat(arr_interfaces_y)
+            c_alpha1 = self.alpha
+
+            fit(handle_[0],
+                <RankSizePair**>rankSizePair,
+                <size_t> n_total_parts,
+                <floatData_t**>data,
+                <size_t>M,
+                <size_t>N,
+                <floatData_t**>labels,
+                <float*>coef_ptr,
+                <float*>&c_intercept1,
+                <bool>self.fit_intercept,
+                <bool>self.normalize,
+                <int>self.max_iter,
+                <float>self.alpha,
+                <float>self.l1_ratio,
+                <bool>self.shuffle,
+                <float>self.tol,
+                False)
```

Comment:
verbose is always set to False, i was wondering if it would help the user if we passed it as a variable